pdearena/data/oned/datapipes/kuramotosivashinsky1d.py

```diff
@@ -0,0 +1,194 @@
+        self.allow_shuffle = allow_shuffle
+        self.dtype = np.float32
+        self.resolution = resolution
+        self.usegrid = usegrid
+        self.time_step = time_step
+        print(f"Loading {mode} data from {len([p for p in dp])} files.")
+        self.storage = {}
+        if preload:
+            for path in self.dp:
+                self.storage[path] = self._load_data(path)
+
+    def _load_data(self, path):
+        if path in self.storage:
+            return self.storage[path]
+        else:
+            with h5py.File(path, "r") as f:
+                data_h5 = f[self.mode]
+                data_key = [
+                    k for k in data_h5.keys() if k.startswith("pde_")][0]
+                data = {
+                    "u": torch.tensor(data_h5[data_key][:].astype(self.dtype)),
+                    "dt": torch.tensor(data_h5["dt"][:].astype(self.dtype)),
+                    "dx": torch.tensor(data_h5["dx"][:].astype(self.dtype)),
+                }
+                if "v" in data_h5:
+                    data["v"] = torch.tensor(
+                        data_h5["v"][:].astype(self.dtype))
+
+                data["orig_dt"] = data["dt"].clone()
+                if data["u"].ndim == 3:
+                    data["u"] = data["u"].unsqueeze(
+                        dim=-2)  # Add channel dimension
+                # Scaling dt to be in the range [0, 10] to be visible changes in fourier embeds
+                if data["dt"].min() > 0.3 and data["dt"].max() < 0.5:
+                    data["dt"] = (data["dt"] - 0.3) * 50.0
+                elif data["dt"].min() > 0.15 and data["dt"].max() < 0.25:
+                    data["dt"] = (data["dt"] - 0.15) * 100.0
+                else:
+                    print(
+                        f"WARNING: dt is not in the expected range (min {data['dt'].min()}, max {data['dt'].max()}, mean {data['dt'].mean()}) - scaling may be incorrect."
+                    )
+                # Scaling dx to be in the range [0, 10] to be visible changes in fourier embeds
+                if data["dx"].min() > 0.4 and data["dx"].max() < 0.6:
+                    data["dx"] = (data["dx"] - 0.4) * 50.0
+                elif data["dx"].min() > 0.2 and data["dx"].max() < 0.3:
+                    data["dx"] = (data["dx"] - 0.2) * 100.0
+                else:
+                    print(
+                        f"WARNING: dt is not in the expected range (min {data['dx'].min()}, max {data['dx'].max()}, mean {data['dx'].mean()}) - scaling may be incorrect."
+                    )
+
+                if "v" in data:
+                    if data["v"].min() >= 0.5 and data["v"].max() <= 1.5:
+                        data["v"] = (data["v"] - 0.5) * 100.0
+                    else:
+                        print(
+                            f"WARNING: v is not in the expected range (min {data['v'].min()}, max {data['v'].max()}, mean {data['v'].mean()}) - scaling may be incorrect."
+                        )
```

Comment:
Shouldn't these have happened in data generation instead?

Reply:
Hm, I had preferred to do them in the data loading since `dt` and `dx` have units that are easier to relate to (i.e. seconds / meters). Rescaling them already in the data removes this relation, especially if someone wants to read them out for other purposes. An alternative would be to add extra keys in the hdf5 file like `scaled_dt` and `scaled_dx` which contain then this calculation (`dt` and `dx` are quite cheap to store). Would that be preferable?

Reply:
Ah, it's alright then. Maybe should add a short note because without context it feels like we are changing the underlying data behind the phenomenon itself to emphasize certain properties.

Reply:
Added a clarification in c96734ba2fd1bd07387d40f603d1e4110f3c7b8b